src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,115 @@
+package com.internet.shop.dao.jdbc;
+
+import com.internet.shop.dao.ProductDao;
+import com.internet.shop.exceptions.DataProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Product;
+import com.internet.shop.util.ConnectionUtil;
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ProductDaoJdbcImpl implements ProductDao {
+    @Override
+    public Product create(Product item) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "INSERT INTO products (name, price) "
+                    + "VALUES (?, ?);";
+            PreparedStatement statement = connection.prepareStatement(query,
+                    PreparedStatement.RETURN_GENERATED_KEYS);
+            statement.setString(1, item.getName());
+            statement.setBigDecimal(2, item.getPrice());
+            statement.executeUpdate();
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            while (generatedKeys.next()) {
+                long productId = generatedKeys.getLong(1);
+                item.setId(productId);
+            }
+            return item;
+        } catch (SQLException e) {
+            throw new DataProcessingException("It is impossible to create a product: "
+                    + item, e);
+        }
+    }
+
+    @Override
+    public Optional<Product> getById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM products "
+                    + "WHERE product_id = ?;";
```

Comment:
```suggestion
            String query = "SELECT * FROM products WHERE product_id = ?;";
```
should we get deleted a product?